src/Common/Factory/FactoryCreateContainOneNewRule.php

```diff
@@ -33,7 +33,7 @@ public function getDescription()
      */
     public function apply(AbstractNode $node)
     {
-        if (!$this->isFactory($node)) {
+        if (!$this->isFactoryExceptPersistence($node)) {
```

Comment:
This method name feels really awkward. I'd have to understand more about the reason why we make an exception for persistence, can you share a bit about that with me?

Reply:
I left all the other factory rules as they was before - excluding persistence factories. We've decided that "only public methods allowed" rule should be also applied to persistence factory so I've created another method and renamed the existing one to avoid ambiguity.

Reply:
From a logic perspective, should we be ignoring persistence methods for this flaw, or is this done to minimize the number of behavior changes?

Reply:
The second one, I guess if someone needed to exclude the persistence factories form all the methods - this should be preserved.

Reply:
I think we can keep it simple.

Reply:
@pushokwhite We can break BC here, this is a tool by us and for us to govern architecture.

Let's keep it at one method, no need to support the `isFactoryExceptPersistence` case.

Reply:
So we need to exclude persistence factories from all the rules (as it was before) including the new one, right?

Reply:
Why not also just saying they only need to allow public methods?
Keep things simple => no exceptions